File: ePrijevozSarajevo/UI/eprijevoz_desktop/lib/models/malfunction.dart
import 'package:json_annotation/json_annotation.dart';
part 'malfunction.g.dart';

@JsonSerializable()
class Malfunction {
  int? malfunctionId;
  String? description;
  DateTime? dateOfMalufunction;
  bool? fixed;
  int? vehicleId;
  int? stationId;

  Malfunction(
      {this.malfunctionId,
      this.description,
      this.dateOfMalufunction,
      this.fixed,
      this.vehicleId,
      this.stationId});

  factory Malfunction.fromJson(Map<String, dynamic> json) =>
      _$MalfunctionFromJson(json);

  Map<String, dynamic> toJson() => _$MalfunctionToJson(this);
}

File: ePrijevozSarajevo/ePrijevozSarajevo.Model/Malfunction.cs
namespace ePrijevozSarajevo.Model
{
    public class Malfunction
    {
        public int MalfunctionId { get; set; }
        public string Description { get; set; } = null!;
        public DateTime? DateOfMalufunction { get; set; }
        public bool? Fixed { get; set; }
        public Vehicle? Vehicle { get; set; } 
        public int? VehicleId { get; set; }
        public Station? Station { get; set; }
        public int? StationId{ get; set; }
    }
}
